tensorflow/contrib/go/tensor.go

```diff
@@ -0,0 +1,763 @@
+package tensorflow
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math"
+	"reflect"
+	"runtime"
+	"unsafe"
+
+	"github.com/golang/protobuf/proto"
+
+	pb "github.com/tensorflow/tensorflow/tensorflow/contrib/go/proto"
+)
+
+import "C"
+
+const (
+	cAckByte  = 6
+	cBellByte = 7
+	cDc1      = 17
+
+	cBytesComplex64 = 8
+	cBytesFloat32   = 4
+	cBytesFloat64   = 8
+	cBytesInt16     = 2
+	cBytesInt32     = 4
+	cBytesInt64     = 8
+	cBytesUint16    = 2
+)
+
+// DataType Type of the data contained by a Tensor
+type DataType pb.DataType
+
+// TensorInt Interface to be implemented by the tensors.
+type TensorInt interface {
+	Data() []byte
+	DataSize() int64
+	DataType() DataType
+	GetVal(d ...int) (val interface{}, err error)
+
+	Dim(n int) int
+	NumDims() int
+
+	AsBool() (res []bool, err error)
+	AsFloat32() (res []float32, err error)
+	AsFloat64() (res []float64, err error)
+	AsInt32() (res []int32, err error)
+	AsInt64() (res []int64, err error)
+	AsStr() (res [][]byte, err error)
+
+	String() string
+}
+
+// Tensor Holds a multi-dimensional array of elements of a single data type.
+type Tensor struct {
+	pb.TensorProto
+
+	tensor      TF_Tensor
+	dimWeights  []int
+	memReleased bool
+}
+
+// TensorShape represents the shapre of a Tensor.
+type TensorShape [][]int64
+
+var (
+	// DtInvalid Invalid tensor DataType.
+	DtInvalid = DataType(0)
```

Comment:
Do you mean the Dt*? I'm not completely sure about this, it makes sense to remove the prefix since I think that have problems with name collisions is unlikely , but in the other hand the prefix makes easy to recognise that this is a DataType when reading the code.
What do you think @dave-andersen ?